src/managers/DeviceManager.js

```diff
@@ -117,11 +114,11 @@ class DeviceManager {
 
     return ({
       ...attributes,
-      connected: true,
-      functions: description.state.f,
+      connected: device && device.ping().connected || false,
+      functions: description ? description.state.f : null,
       lastFlashedAppName: null,
-      lastHeard: new Date(),
-      variables: description.state.v,
+      lastHeard: device && device.ping().lastPing || attributes.lastHeard,
```

Comment:
Nice catch. with that pr: https://github.com/Brewskey/spark-protocol/pull/78, I added updating lastHeard in attributes onDeviceReady and onDeviceDisconnect.. and here, if a device is connected we get lastPing, if no, we get lastHeard from attributes.